### PR TITLE
external links in rich text blocks now opening in new tab

### LINKED ISSUE #8002: Update rich text streamfield blocks to open external links in a new tab
Follow up from - [PNI] Make external links open in a new tab #7878 

We've started testing to see if opening external links in a new tab affects our analytics around bounce rate and sessions in a positive way. tl;dr is yes so we've have decided to expand this to include universal new-tab link behaviour from the rich text streamfield component in order to update this across the entire CMS.

- [x] All rich text streamfield blocks open external links in a new tab

### PR DESCRIPTION
**Pull request by danielfmiranda**
Closes #8002 

**Link to sample test page**: https://foundation-s-rich-text--1wiazw.mofostaging.net/en/blog/initial-test-blog-post-with-fixed-title/

**Steps to test**:

1. Please visit the test page above 
2. In the posts content, please click on any of the external links such as the ones that say "This is a link to a fake URL!" or the ones found in the image or quote block.
3. These all should open to a new tab
4. If you would like to test another page type, visit https://foundation-s-rich-text--1wiazw.mofostaging.net/en/publication-page-with-child-article-pages/fixed-title-article-page/ to test out the external rich text links found there.
5. If everything is working as expected, testing is complete!

### CODE REVIEW

File: network-api/networkapi/wagtailpages/wagtail_hooks.py [hunk 1]
```diff
@@ -84,6 +86,21 @@ def register_large_feature(features):
     features.default_features.append('large')
 
 
+# Updating external links in rich text blocks to open in a new tab
+class RichTextExternalLinkNewTabHandler(LinkHandler):
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/wagtail_hooks.py:90]: Docs Referenced: https://docs.wagtail.org/en/latest/extending/rich_text_internals.html#registering-rewrite-handlers

**Review by Pomax — APPROVED** [review]
This looks good, although it of course only works for richtext links so we may be setting up inconsistent behgavuour for our users.

e.g.  all links open in a new window, but not this one, because it's not a richtext link:

image

**Comment by kristinashu** [comment]
Thank you for setting up the examples Daniel! The link types you listed are all working well. But I did notice the external links in the captions and the external button link are still opening in the same tab. Is that intentional, is it possible to make them open in a new tab as well? 

image

**Comment by danielfmiranda** [comment]
Hi @kristinashu! That was intentional, from my understanding, the ticket asked for only links in rich text fields.
However if you would prefer that all external links or a specific list of link types on the site open in a new tab, that is definitely doable as well! 

Should I create a new ticket for that or should I add it to this ticket? 

Thank you!

**Comment by kristinashu** [comment]
Ah ok thank you for explaining. Yes, perhaps open a new ticket to tackle any remaining external links. Will approve this one now tho!

**Review by kristinashu — APPROVED** [review]

**Comment by danielfmiranda** [comment]
Sounds good @kristinashu thanks for the approval! 

I have created https://github.com/mozilla/foundation.mozilla.org/issues/8214 to tackle the rest of the links on the page 😃

**Pull request merged**